### PR TITLE
solution

### PR DESCRIPTION
**Pull request by tarasavage**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,8 +1,23 @@
...
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list) -> list[Person]:
+    married_people = []
+    for person in people:
+        new_person = Person(person["name"], person["age"])
+        if "wife" in person.keys() and person["wife"]:
```
Comment (Grayder0152) [inline app/main.py:14]: Instead of this condition you can use `person.get('wife') is not None`

Comment (Grayder0152) [reply app/main.py:14]: Apart from this, you can use `:=` operator, but it is not necessary

File: app/main.py [hunk 2]
```diff
@@ -1,8 +1,23 @@
...
+            wife = Person(person["name"], person["age"])
+            wife.husband = new_person
+            new_person.wife = wife
+        elif "husband" in person.keys() and person["husband"]:
```
Comment (Grayder0152) [inline app/main.py:18]: The same as above

File: app/main.py [hunk 3]
```diff
@@ -1,8 +1,23 @@
...
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list) -> list[Person]:
+    married_people = []
+    for person in people:
+        new_person = Person(person["name"], person["age"])
+        if "wife" in person.keys() and person["wife"]:
+            wife = Person(person["name"], person["age"])
```
Comment (Grayder0152) [inline app/main.py:15]: You need to get person from `people` not create

**Review by Grayder0152 — APPROVED** [review]

File: app/main.py [hunk 4]
```diff
@@ -1,8 +1,23 @@
...
+            wife.husband = new_person
+            new_person.wife = wife
+        elif "husband" in person.keys() and person["husband"]:
+            husband = Person(person["name"], person["age"])
```
Comment (Grayder0152) [inline app/main.py:19]: The same as above

**Review by Grayder0152 — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 5]
```diff
@@ -1,8 +1,23 @@
...
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(people: list) -> list[Person]:
```
Comment (Bohdan-Salamakha) [inline app/main.py:10]: add additional typehint for list inner objects

File: app/main.py [hunk 6]
```diff
@@ -1,8 +1,23 @@
...
+    married_people = [
+        Person(person["name"], person["age"])
```
Comment (Bohdan-Salamakha) [inline app/main.py:12]: use get method instead of parenthesis everywhere to avoid KeyError

**Review by Bohdan-Salamakha — CHANGES_REQUESTED** [review]
v1

**Pull request closed without merging**